subsys/random/rand32_entropy_device.c

```diff
@@ -42,3 +43,74 @@ u32_t sys_rand32_get(void)
 
 	return random_num;
 }
+
+static void rand_get(u8_t *dst, u32_t outlen)
+{
+	struct device *dev = entropy_driver;
+	u32_t random_num;
+	int ret;
+
+	if (unlikely(!dev)) {
+		/* Only one entropy device exists, so this is safe even
+		 * if the whole operation isn't atomic.
+		 */
+		dev = device_get_binding(CONFIG_ENTROPY_NAME);
+		__ASSERT((dev != NULL),
+			"Device driver for %s (CONFIG_ENTROPY_NAME) not found. "
+			"Check your build configuration!",
+			CONFIG_ENTROPY_NAME);
+		entropy_driver = dev;
+	}
+
+	ret = entropy_get_entropy(dev, dst, outlen);
+
+	if (unlikely(ret < 0)) {
+		/* Use system timer in case the entropy device couldn't deliver
+		 * 32-bit of data.  There's not much that can be done in this
+		 * situation.  An __ASSERT() isn't used here as the HWRNG might
+		 * still be gathering entropy during early boot situations.
+		 */
+
+		u32_t len = 0;
+		u32_t blocksize = 4;
+
+		while (len < outlen) {
+			random_num = k_cycle_get_32();
+			if ((outlen-len) < sizeof(random_num)) {
+				blocksize = len;
+				(void *)memcpy(&(dst[random_num]),
+						&ret, blocksize);
+			} else {
+				*((u32_t *)&dst[len]) = random_num;
+			}
+
+			len += blocksize;
+		}
+	}
+}
+
+#if CONFIG_ENTROPY_DEVICE_RANDOM_GENERATOR
+u32_t sys_rand32_get(void)
+{
+	return rand32_get();
+}
+
+void sys_rand_get(void *dst, u32_t outlen)
+{
+	return rand_get(dst, outlen);
+}
+#endif /* CONFIG_ENTROPY_DEVICE_RANDOM_GENERATOR */
+
+#if CONFIG_ENTROPY_DEVICE_CS_GENERATOR
```

Comment:
better use `#if defined(...)` to avoid undefined macro warnings
(also everywhere else where you added a new #if)

Reply:
Ok